build_shared_vars.sh

```diff
@@ -0,0 +1,63 @@
+find_repo_root()
```

Comment:
> I wasn't intending to go "this far" with root detection, just a `$(realpath ../../../..)` would have been enough for me if `ANDROID_BUILD_TOP` isn't set, but such a function is definitely a nice addition.
> Anyway I don't like the arbitrary depth number, when a path is guaranteed to have a root somewhere. Nor `cd` ing, but that's mostly personal preference.
I added a limit to prefent the function from jumping outside of the repo tree if its broken into another one (I have to for example)

Reply:
> Replacing the `dirname` invocation with `${2%/*}` (since dirname doesn't do any canonicalization, just _string_ manipulations), and perhaps `test` with `[ ]` or `[[ ]]` seems fine to me.
Do you think there are cases were ${0} is either not something like ./path/to/script or ./script?
As long as the path is absolute its not important.

Reply:
> I added a limit to prefent the function from jumping outside of the repo tree if its broken into another one (I have to for example)

Then it should anyway find the nested folder containing `.repo`, otherwise different problems arise. 10 folders is a lot, yet arbitrary.

> Do you think there are cases were ${0} is either not something like ./path/to/script or ./script?
> As long as the path is absolute its not important.

`"$PWD"` is always set to an absolute path afaik.

Reply:
But I was talking about $0 thats what you were talking about. I think this simple solution  is enough here.

Reply:
>Then it should anyway find the nested folder containing .repo, otherwise different problems arise. 10 folders is a lot, yet arbitrary.

It is arbitrary as I think there needs to be a limit that makes sense but also isn't limited when the depth of the script in the tree changes.

Reply:
`${2%/*}` instead of calling `$(dirname $2)`? I think either is fine. Try `$(dirname .)` or `$(dirname ..)` or `$(dirname ../../..)`, you'll see what I mean :)